src/UseCases/SignUpUseCase.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\UseCases;
+
+use App\Domain\User;
+use App\Exceptions\UserAlreadyExistException;
+use Doctrine\ORM\EntityManager;
+
+class SignUpUseCase
+{
+    
+    /**
+     * @var EntityManager
+     */
+    private $repositori;
+
+        public function __construct (UserRepository $repositori)
+    {
+        $this->repositori = $repositori;
+        
+    }
+        public function __invoke(string $username, string $email, string $password)
+    {
+         $user = $this->em->getRepository(User::class)->findOneBy([
```

Comment:
La propiedad `em` ya no existe, ahora debería ser simplemente `$this->repository` en vez de `$this->em->getRepository(User::class)`